### PR TITLE
various fixes for install and security etc

### PR DESCRIPTION
**Pull request by schuermannator**
security/reproducibility fixes including upgrading packages, makefile cleanup, and readme additions to ensure you can build certain required libraries from source (which requires install of `gcc` and some libs)

### CODE REVIEW

**Comment by schuermannator** [comment]
Unfortunately, this PR likely won't address #32. That is going to require some attention to come up with a good fix. I tried replacing with stock `react-select` (v3) but was much too slow.

**Comment by schuermannator** [comment]
currently: `found 106 vulnerabilities (88 low, 16 moderate, 2 high)`  

All can be fixed if we migrate off react-select-2

**Comment by schuermannator** [comment]
good to review

**Comment by samjett247** [comment]
> Unfortunately, this PR likely won't address #32. That is going to require some attention to come up with a good fix. I tried replacing with stock `react-select` (v3) but was much too slow.

Shoot, I'll look for some alternatives.

File: install.sh [hunk 1]
```diff
@@ -3,5 +3,10 @@
 command -v node >/dev/null 2>&1 || { echo >&2 "Nodejs is not installed.  Aborting."; exit 1; }
 command -v npm >/dev/null 2>&1 || { echo >&2 "NPM is not installed.  Aborting."; exit 1; }
 
-echo "Installing required packages..."
+echo "Checking system-level dependencies..."
```
Comment (samjett247) [inline install.sh]: Nice I dig it. V professional.

**Comment by schuermannator** [comment]
vulnerabilities are done with npm. You can run `npm audit` to check

**Pull request merged**